src/FakeItEasy/Core/OrderedFakeAsserter.cs

```diff
@@ -2,15 +2,13 @@ namespace FakeItEasy.Core
 {
     using System;
     using System.Collections.Generic;
-    using System.Globalization;
-    using System.IO;
 
     internal class OrderedFakeAsserter : IFakeAsserter
     {
         private readonly IEnumerable<IFakeObjectCall> originalCallList;
         private readonly CallWriter callWriter;
         private readonly Queue<IFakeObjectCall> calls;
-        private readonly List<AssertedCall> assertedCalls = new List<AssertedCall>();
+        private readonly SynchronizedCollection<AssertedCall> assertedCalls = new SynchronizedCollection<AssertedCall>();

```

Comment:
Well, in any case, I won't take the decision alone, so don't rush to revert your changes. I'd like to have the opinion of other @FakeItEasy/owners too.